src/server/activeobjectmgr.cpp

```diff
@@ -34,6 +34,12 @@ ActiveObjectMgr::~ActiveObjectMgr()
 	}
 }
 
+void ActiveObjectMgr::clear()
+{
+	::ActiveObjectMgr<ServerActiveObject>::clear();
+	m_spatial_map.removeAll();
+}
+
 void ActiveObjectMgr::clearIf(const std::function<bool(ServerActiveObject *, u16)> &cb)
```

Comment:
This method directly removes entries from `m_active_objects`, side-stepping `removeObject`. I believe you need to replace `m_active_objects.remove(it.first);` with `removeObject(it.first);` to ensure that you don't end up with dead entries in your spatial index data structure. At least that's what I did.